main.c

```diff
@@ -67,7 +76,12 @@ int main(void)
             default: break; // 'C_none' does nothing
         }
         Random_Update();
-        clock_update();
+        uint32_t time_now = HAL_GetTick(); // for running a 1ms-interval tick

```

Comment:
would it make sense to just run this on a hardware timer at 1ms? or did we run out of hw timers?

Reply:
all the HW timers are assigned to the metro module (there's 8 available). i think it's best to leave them here, because the priority is allowed to slip if the CPU usage increases. As it is though, we're getting >4000 event loops per millisecond, so i think we're in good shape!